cmd/machine-controller/main.go

```diff
@@ -253,10 +258,23 @@ func main() {
 	var registryMirrors []string
 	for _, mirror := range strings.Split(nodeRegistryMirrors, ",") {
 		if trimmedMirror := strings.TrimSpace(mirror); trimmedMirror != "" {
+			if !strings.HasPrefix(mirror, "http") {
+				trimmedMirror = "https://" + mirror
+			}
+
+			_, err := url.Parse(trimmedMirror)
+			if err != nil {
+				klog.Fatalf("incorrect mirror provided: %v", err)
+			}
+
 			registryMirrors = append(registryMirrors, trimmedMirror)
 		}
 	}
 
+	if len(registryMirrors) > 0 {
+		nodeContainerdRegistryMirrors["docker.io"] = registryMirrors
+	}
```

Comment:
Should we have some validation to fail starting MC if both `registryMirrors` and `nodeContainerdRegistryMirrors["docker.io"]` are provided by the user?

Reply:
No, if docker.io is empty it's OK for dockerd

Reply:
I'm not sure I get it, but isn't one flag going to override another one? For example, if you have both `--node-registry-mirrors` and `--node-containerd-registry-mirrors`, isn't only one going to have effect?

Reply:
Essentially it will, but I don't see it as a problem.

Reply:
It's kind of backward compatibility, it's possible to configure docker and containerd at the same time (of course only 1 will be used, according to -node-container-runtime).

Reply:
Okay, that makes sense, thanks for the explanation!